crates/rpc/rpc-eth-types/src/cache/mod.rs

```diff
@@ -201,6 +205,27 @@ impl<B: Block, R: Send + Sync> EthStateCache<B, R> {
         let _ = self.to_service.send(CacheAction::GetHeader { block_hash, response_tx });
         rx.await.map_err(|_| ProviderError::CacheServiceUnavailable)?
     }
+
+    /// Retrieves a chain of connected blocks from the cache, starting from the given block hash
+    /// and traversing down through parent hashes. Returns blocks in descending order (newest
+    /// first).
+    /// This is useful for efficiently retrieving a sequence of blocks that might already be in
+    /// cache without making separate database requests.
+    /// If the starting block is not in cache, an empty vector is returned.
+    pub async fn get_cached_parent_blocks(
+        &self,
+        block_hash: B256,
+        max_blocks: usize,
+    ) -> Vec<Arc<RecoveredBlock<B>>> {
+        let (response_tx, rx) = oneshot::channel();
+        let _ = self.to_service.send(CacheAction::GetCachedParentBlocks {
+            block_hash,
+            max_blocks,
+            response_tx,
+        });
+
+        rx.await.unwrap_or_default()
```

Comment:
I'd like to return an option here if the vec is empty just so that we can represent this case properly and can guarantee that if this returns a vec it is not empty